pkg/util/conc/future.go

```diff
@@ -31,7 +31,7 @@ type Future[T any] struct {
 	err   error
 }
 
-func newFuture[T any]() *Future[T] {
+func NewFuture[T any]() *Future[T] {
```

Comment:
We made this private to avoid abuse, to guide people to create future by pool (to avoid spawning goroutine directly), it would be better to add a util method:
```
func Error(err error) *Future[T] {
    future := newFuture[T]()
    future.err = err
    close(future.ch)
    return future
}
```

Reply:
fix that with a new call NewErrFuture